### PR TITLE
45 esb architecture

### LINKED ISSUE #45: Improve ElementSelectionBehaviour architecture
We talked about the architecture with Bastiaan, and this is (most likely) going to be the new architecture for ESBs. Here, we don't need an interface over the constructor, but over a static `create` or `creator` method. If TypeScript is tolerant enough, we can instantiate a `name` of type `Class<ESB>` through `name.create(database)`.

ElementSelectionBehaviour architecture

### PR DESCRIPTION
**Pull request by MathiasMeuleman**
Will close #45 

This pull request fixes some inconsistencies and duplicate code in the ElementSelectionBehaviour classes. There where 6 abstract classes that all did exactly the same, these were replaced by a single abstract class called AbstractElementSelectionBehaviour. This abstract class implements the ElementSelectionBehaviour interface. All the ElementSelectionBehaviour subclasses now inherit from the single abstract class.

Also the constructors of the subclasses have been replaced with a single super(DB, ElementID, ElementDsc) call. This removes a lot of duplicate code.

### CODE REVIEW

File: src/octopeer-github/content/ElementSelectionBehaviour/NameElementSelectionBehaviour/OtherContributerNameElementSelectionBehaviour.ts [hunk 1]
```diff
@@ -1,17 +1,15 @@
...
 
     /**
      * Constructor of this class.
      * @param database The database that is used to store the Elements.
      */
     public constructor(database: DatabaseAdaptable) {
-        super(database);
-        this.elementDsc = "teststss";
-        this.elementID = new ElementID(305);
+        super(database, new ElementID(305), "teststss");
```
Comment (mpsijm) [inline src/octopeer-github/content/ElementSelectionBehaviour/NameElementSelectionBehaviour/OtherContributerNameElementSelectionBehaviour.ts]: While you're at it, maybe add a `// TODO` at all the "tesetstststst" things so we can search the source folder for todo's (or use WebStorm's TODO viewer) :)

**Comment by mpsijm** [comment]
There are not many changes that have a lot of impact, and the changes that are there look good to me (except for my last comment) :)

**Comment by mpsijm** [comment]
Ready for merge now! :D

**Comment by thervh70** [comment]
Looks great. An easy way to boost test coverage :P.

Will merge!

**Pull request merged**